src/arrangement/core.cljc

```diff
@@ -78,10 +90,11 @@
         (compare-seqs rank a b)
 
         :else
-        (let [class-diff (compare (.getName (class a))
-                                  (.getName (class b)))]
+        (let [class-diff (compare (type-name a)
+                                  (type-name b))]
           (if (zero? class-diff)
-            (if (instance? java.lang.Comparable a)
-              (compare a b)
-              (compare (pr-str a) (pr-str b)))
+            #?(:clj (if (instance? java.lang.Comparable a)
+                      (compare a b)
+                      (compare (pr-str a) (pr-str b)))
+               :cljs (compare a b))
```

Comment:
What happens if the types here don't implement `IComparable`? You could change this so that the reader conditional is around the test in the `if`.